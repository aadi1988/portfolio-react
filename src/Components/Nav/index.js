import React from 'react'
import { capitalizeFirstLetter } from "../../utils/helpers";

function Nav(props){
   const {
      categories = [],
      setCurrentCategory,
      currentCategory,
      portfolioSelected,
      setPortfolioSelected
    } = props;
 
   return(
      
      <nav>
         <ul className="NavUl">
             <li className="NavLi">
                <a href="#" onClick={() => setPortfolioSelected(false)}>
                   Cover
                </a>
              </li>
            {categories.map((category) => (
                <li className={`NavLi ${currentCategory === category && !portfolioSelected && 'navActive'}`} key={category}>
                  <span onClick={() => {setCurrentCategory(category);setPortfolioSelected(true);}}>
                    {capitalizeFirstLetter(category)}
                  </span>
                </li>
              ))}
         </ul>
      </nav>
    
   );
    
}

export default Nav;