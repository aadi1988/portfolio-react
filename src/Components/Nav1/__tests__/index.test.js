import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';
const categories = ['Portfolio']
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

afterEach(cleanup);
describe('Nav component', () => {
    // baseline test
    it('renders', () => {
        render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
          />);
    });
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav
            categories={categories}
            setCurrentCategory={mockSetCurrentCategory}
            currentCategory={mockCurrentCategory}
          />);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('links are visible', () => {
    it('inserts text into the links', () => {
      // Arrange
      const { getByTestId } = render(<Nav
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
      />);
      // Assert
      expect(getByTestId('cover')).toHaveTextContent('Cover');
      
    });
})