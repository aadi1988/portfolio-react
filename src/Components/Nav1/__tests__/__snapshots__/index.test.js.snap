// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark"
  >
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="mx-2 nav-link"
          href="#"
          role="button"
        />
        <a
          data-testid="cover"
          href="#"
        >
          Cover
        </a>
        <a
          class="false nav-link"
          href="#"
          role="button"
        >
          <span>
            Portfolio
          </span>
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
